virtual_rainforest/models/abiotic_simple/simple_regression.py

```diff
@@ -0,0 +1,677 @@
+    reference_data: DataArray,
+    leaf_area_index_sum: DataArray,
+    layer_roles: List,
+    layer_heights: DataArray,
+    upper_bound: float,
+    lower_bound: float,
+    gradient: float,
+) -> DataArray:
+    """LAI regression and logarithmic interpolation of variables for vertical profile.
+
+    Args:
+        data: Data object
+        reference_data: input variable at reference height
+        leaf_area_index_sum: leaf area index summed over all layers, [m m-1]
+        layer_roles: list of layer roles (soil, surface, subcanopy, canopy, above)
+        layer_heights: vertical layer heights, [m]
+        lower_bound: minimum allowed value
+        upper_bound: maximum allowed value
+        gradient: gradient of regression from :cite:t:`hardwick_relationship_2015`
+
+    Returns:
+        vertical profile of provided variable
+    """
+
+    # Calculate microclimatic variable at 1.5 m as function of leaf area index
+    lai_regression = DataArray(
+        leaf_area_index_sum * gradient + reference_data, dims="cell_id"
+    )
+
+    # Calculate per cell slope and intercept for logarithmic within-canopy profile
+    slope = (reference_data - lai_regression) / (
+        np.log(layer_heights.isel(layers=0)) - np.log(1.5)
+    )
+    intercept = lai_regression - slope * np.log(1.5)
+
+    # Calculate the values within cells by layer
+    layer_values = np.log(layer_heights) * slope + intercept
+
+    # set upper and lower bounds
+    layer_values_low = np.where(layer_values < lower_bound, lower_bound, layer_values)
+    return DataArray(
+        np.where(layer_values_low > upper_bound, upper_bound, layer_values_low),
+        dims=["layers", "cell_id"],
+        coords={
+            "layers": np.arange(0, len(layer_roles)),
+            "layer_roles": ("layers", layer_roles[0 : len(layer_roles)]),
+            "cell_id": data.grid.cell_id,
+        },
+    )
+
+
+def calculate_saturation_vapor_pressure(
+    temperature: DataArray,
+    factor1: float = MicroclimateParameters["saturation_vapor_pressure_factor1"],
+    factor2: float = MicroclimateParameters["saturation_vapor_pressure_factor2"],
+    factor3: float = MicroclimateParameters["saturation_vapor_pressure_factor3"],
+) -> DataArray:
+    r"""Calculate saturation vapor pressure.
+
+    Saturation vapor pressure :math:`e_{s} (T)` is here calculated as
+    :math:`e_{s}(T) = 0.61078 exp(\frac{7.5 T}{T + 237.3})`
+    where :math:`T` is temperature in degree C .
+
+    Args:
+        temperature: air temperature, [C]
+        factor1: factor 1 in saturation vapor pressure calculation
+        factor2: factor 2 in saturation vapor pressure calculation
+        factor3: factor 3 in saturation vapor pressure calculation
+
+    Returns:
+        saturation vapor pressure, kPa
+    """
+    return DataArray(
+        (factor1 * np.exp((factor2 * (temperature)) / (temperature + factor3))),
+        dims=temperature.dims,
+        name="saturation_vapor_pressure",
+    )
```

Comment:
As in another case, I believe you can simply do:
```suggestion
    return (factor1 * np.exp(factor2 * temperature) / (temperature + factor3)).rename("saturation_vapor_pressure")
    )
```
Also, maybe `"saturation_vapour_pressure"`?

Reply:
This again is a mypy issue, but I found a compromise